### PR TITLE
First pass at enabling structs, arrays, and maps for more parts of the plan

### PR DESCRIPTION
**Pull request by revans2**
This covers a lot of the high level SparkPlan parts.  There are follow on issues for joins and several other parts where I ran into a few issues. I will keep working on expressions and other operators, but those are more likely to be follow on issues instead of fixes that can be done right away because most of the time we already support about as much as cudf allows us to.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: integration_tests/src/main/python/window_function_test.py [hunk 1]
```diff
@@ -838,3 +837,16 @@ def do_it(spark):
...
 
     assert_gpu_fallback_collect(do_it, 'WindowExec')
+
+# In a distributed setup the order of the partitions returend might be different, so we must ignore the order
+# but small batch sizes can make sort very slow, so do the final order by locally
+@ignore_order(local=True)
+@pytest.mark.parametrize('ride_along', all_basic_gens + decimal_gens + array_gens_sample + struct_gens_sample + map_gens_sample, ids=idfn)
+def test_window_ride_along(ride_along):
+    assert_gpu_and_cpu_are_equal_sql(
+            lambda spark : gen_df(spark, [('a', LongRangeGen()), ('b', ride_along)]),
```
Comment (jlowe) [inline integration_tests/src/main/python/window_function_test.py:847]: Nit: indentation appears to be off here relative to other parameters being passed.

**Review by jlowe — DISMISSED** [review]
Minor indentation nit but otherwise lgtm.

**Comment by revans2** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**